### PR TITLE
Findthebreak

### PR DESCRIPTION
**Pull request by tessbianchi**
- Adds changes to videoplayer so that it can have a basic mode

- adds ftb mission

- some of this code is untested, I will not merge without testing, but anthony wanted to see some of this code.

### CODE REVIEW

File: mission_control/navigator_missions/nav_missions/find_the_break.py [hunk 1]
```diff
@@ -1,12 +1,79 @@
...
 @txros.util.cancellableInlineCallbacks
 def main(navigator, **kwargs):
-    fprint("STARTING FIND THE BREAK, GETTING A BOUY AS A TEST", msg_color="green")
+    # mission params
+    meters_between_buoys = 20
+    radius = 40
+
+    # var initialization
+    nh = navigator.nh
+    test = False
```
Comment (mattlangford) [inline mission_control/navigator_missions/nav_missions/find_the_break.py]: For dictionaries, you can do `test = kwargs.get("test", False)` to get it if it's in the dictionary and default to False if it's not

File: mission_control/navigator_missions/nav_missions/find_the_break.py [hunk 2]
```diff
@@ -1,12 +1,79 @@
...
+    else:
+        perception = yield FindTheBreakPerception(nh).init_()
+    dbhelper = yield DBHelper(nh).init_(navigator)
+
+    fprint("STARTING FIND THE BREAK", msg_color="green")
     # Get all objects within a certain radius of type unknown or buoy
+    nav_pos = yield navigator.tx_pose
+    nav_pos = nav_pos[0]
+    objects = yield dbhelper.get_objects_in_radius(nav_pos, radius, ["buoy", "unknown"])
+    if len(objects) < 2:
```
Comment (mattlangford) [inline mission_control/navigator_missions/nav_missions/find_the_break.py]: Why less than 2?

File: mission_control/navigator_missions/nav_missions/find_the_break.py [hunk 3]
```diff
@@ -1,12 +1,79 @@
...
+        fprint("No objects in proximity", msg_color="red")
+        raise Exception("No objects in proximity")
+
     # Get all combinations and find the ones with the correct distance
+    objects_comb = itertools.combinations(objects, 2)
+    min_dist = sys.maxint
```
Comment (mattlangford) [inline mission_control/navigator_missions/nav_missions/find_the_break.py]: np.inf

File: mission_control/navigator_missions/nav_missions/find_the_break.py [hunk 4]
```diff
@@ -1,12 +1,79 @@
...
+    min_objs = None
+    for o in objects_comb:
+        dist = np.linalg.norm(nt.rosmsg_to_numpy(o[0].position) - nt.rosmsg_to_numpy(o[1].position))
+        if abs(dist - meters_between_buoys) < min_dist:
```
Comment (mattlangford) [inline mission_control/navigator_missions/nav_missions/find_the_break.py]: This line is confusing to me

File: mission_systems/navigator_find_the_break/find_the_break_perception.py [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+        image_as_centers = image_as_centers.reshape((frame.shape))
+        labels = label.reshape((frame.shape[:2]))
+
+        # Get the bounding boxes of the clusters
+        rects = []
+
+        for i in range(0, K):
+            ind = np.where(labels == i)
+            points = ind
+            points = [[points[1], points[0]]]
```
Comment (mattlangford) [inline mission_systems/navigator_find_the_break/find_the_break_perception.py]: No need to do a 2D list here I think, if you just do `np.array([points[1], points[2])` you should be swell.

File: mission_systems/navigator_find_the_break/find_the_break_perception.py [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+        self.debug.add_image(draw1, "real_rects", topic="real_rects")
+        return hrects, vrects
+
+    def _update_pipes(self, pipes, old_pipes):
+        new_pipes = []
+        if len(old_pipes) == 0:
+            return pipes
+        for i, h in enumerate(pipes):
+            dists = map(lambda x: np.linalg.norm(h - x), old_pipes)
+            pipe_to_update = min(old_pipes, key=lambda x: np.linalg.norm(h - x))
```
Comment (mattlangford) [inline mission_systems/navigator_find_the_break/find_the_break_perception.py]: The argmin of dists is the idx

File: utils/navigator_tools/nodes/video_player [hunk 7]
```diff
@@ -28,102 +28,123 @@ import rospy
...
+            cv2.createTrackbar('Frame', 'Video Control', int(self.start_frame), int(self.num_frames), self.trackbar_cb)
+            cv2.createTrackbar('ROI x_offset', 'Video Control', int(self.roi_x_offset), int(self.width) - 1, self.roi_x_cb)
+            cv2.createTrackbar('ROI y_offset', 'Video Control', int(self.roi_y_offset), int(self.height) - 1, self.roi_y_cb)
+            cv2.createTrackbar('ROI height', 'Video Control', int(self.roi_height), int(self.height), self.roi_height_cb)
+            cv2.createTrackbar('ROI width', 'Video Control', int(self.roi_width), int(self.width), self.roi_width_cb)
+        rospy.loginfo("Playing {} at {}fps starting at frame {} ({} Total Frames)".format(
+            self.filename, self.fps, self.start_frame, self.num_frames))
+
+    def run(self):
+        r = rospy.Rate(self.fps)  # 10hz
```
Comment (mattlangford) [inline utils/navigator_tools/nodes/video_player]: A wrong comment here

**Comment by whispercoros** [comment]
Depricated

**Pull request closed without merging**